dist/index.js

```diff
@@ -38166,58 +38226,101 @@ const core = __nccwpck_require__(2153);
 const fs = __nccwpck_require__(7147);
 const yaml = __nccwpck_require__(3966);
 const path = __nccwpck_require__(1017);
-const { componentYamlSchemaV1D0 } = __nccwpck_require__(1677);
+const {
+  componentYamlSchemaV1D0,
+  endpointYamlSchemaV0D1,
+  componentConfigYamlSchemaV1beta1,
+} = __nccwpck_require__(1677);
+
+const SourceConfigFileTypes = {
+  COMPONENT_YAML: "component.yaml",
+  COMPONENT_CONFIG_YAML: "component-config.yaml",
+  ENDPOINT_YAML: "endpoints.yaml",
+};
 
 function readInput() {
-  try {
-    sourceRootDir = core.getInput("source-root-dir-path");
-    return sourceRootDir;
-  } catch (error) {
-    throw new Error(`Failed to read input: ${error.message}`);
-  }
+  sourceRootDir = core.getInput("source-root-dir-path");
+  fileType = core.getInput("file-type");
+  return [sourceRootDir, fileType];
 }
 
-function readComponentYaml(filePath) {
+function readSrcConfigYaml(filePath, fileType) {
   try {
-    fullPath = path.join(filePath, ".choreo", "component.yaml");
+    let fullPath = path.join(filePath, ".choreo");
+    switch (fileType) {
+      case SourceConfigFileTypes.COMPONENT_YAML:
+        fullPath = path.join(fullPath, SourceConfigFileTypes.COMPONENT_YAML);
+        break;
+      case SourceConfigFileTypes.COMPONENT_CONFIG_YAML:
+        fullPath = path.join(
+          fullPath,
+          SourceConfigFileTypes.COMPONENT_CONFIG_YAML
+        );
+        break;
+      case SourceConfigFileTypes.ENDPOINT_YAML:
+        fullPath = path.join(fullPath, SourceConfigFileTypes.ENDPOINT_YAML);
+        break;
+      default:
+        throw new Error(`'${fileType}' is not a valid source config file type`);
+    }
     let fileContent = fs.readFileSync(fullPath, "utf8");
     return fileContent;
   } catch (error) {
-    throw new Error(`Failed to read component.yaml: ${error.message}`);
+    throw new Error(`\nFailed to read source config file: ${error.message}`);
```

Comment:
Why we add \n here?

Reply:
This is needed as when this gets appended to the final err msg, this should start in a new line